pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -232,6 +232,11 @@ def on_run_end(self) -> None:
         if self._num_training_batches_reached(self.is_last_batch):
             self.update_lr_schedulers("epoch", update_plateau_schedulers=True)
 
+        # on a complete run, reset current progress counters in case we want to resume
+        self.batch_progress.current.reset()
+        self.scheduler_progress.current.reset()
+        self.batch_loop.optimizer_loop.optim_progress.reset_on_epoch()
+
```

Comment:
I'm not sure about this. Wouldn't this break subsequent `trainer.fit` calls where the first one didn't finish?

why can't we put it in `def reset`?

Reply:
sorry, I need to take this PR back and think about this more, mainly for other reasons but yes you have also a good point. My intention is to fix the problem where the checkpoint contains progress values that would make the loop restart on the last batch of the previous epoch. 

We are missing the logic in progress tracking to resume on the next batch, when so far we have only focused on resuming on the current batch if it failed.